gschem/src/x_newtext.c

```diff
@@ -292,7 +293,10 @@ static void newtext_init(NewText *dialog)
                     0);
 
   dialog->textsizecb = gschem_integer_combo_box_new();
-  gschem_integer_combo_box_set_value(dialog->textsizecb, 12);
+
+  gschem_integer_combo_box_set_value (dialog->textsizecb,
+                                      g_current_window()->text_size);
```

Comment:
Please don't use third party objects in the dialog init function. It should add initial values that apply to the dialog only. Other values you can set on new `gobject` instance of the dialog.

Please don't use Scheme extension code inside dialog C code. Here I mean `g_current_window`. I and some devs don't want to mix it.

Since we don't have appropriate properties for text size in `NewText` yet (which has to be fixed in future, I think), you can do the same direct assigning in the function `text_input_dialog()` where new dialog objects are instantiated. For example:
> gschem_integer_combo_box_set_value (NEWTEXT (w_current->tiwindow)->textsizecb,
>   w_current->text_size);